openassessment/xblock/submission_mixin.py

```diff
@@ -75,8 +75,10 @@ def submit(self, data, suffix=''):  # pylint: disable=unused-argument
             suffix (str): Not used in this handler.
 
         Returns:
-            (tuple): Returns the status (boolean) of this request, the
+            (tuple | [tuple]): Returns the status (boolean) of this request, the
                 associated status tag (str), and status text (unicode).
+                This becomes an array of similarly structured tuples in the event
```

Comment:
A thought for future refactoring: should this method always return a list of tuples, to make processing more consistent from the client's perspective?

Reply:
Admittedly, the result is surfaced to the client but never actively read/used. I'd be even happier to return a dict or something more useful than a tuple so that it would be more helpful if the client ever decided to use that info.